From: Bruno Thomsen <bth@kamstrup.dk>
Date: Thu, 25 Sep 2014 15:52:52 +0200
Subject: [PATCH] bash-3.2: patch 51

Bash-Release: 3.2
Patch-ID: bash32-051

Bug-Reported-by:	werner@suse.de
Bug-Reference-ID:	<201002251238.o1PCcYcg016893@boole.suse.de>
Bug-Reference-URL:	http://lists.gnu.org/archive/html/bug-bash/2010-02/msg00132.html

Bug-Description:

When the `read' builtin times out after the timeout specified with -t is
exceeded, it does not reset the flags that tell signal handlers to process
signals immediately instead of deferring their handling.  This can result
in unsafe functions being called from signal handlers, which can cause bash
to hang or dump core.

Signed-off-by: Bruno Thomsen <bth@kamstrup.dk>
---
 builtins/read.def | 2 ++
 patchlevel.h      | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/builtins/read.def b/builtins/read.def
index afa549ed9d42..1c18c8d9e039 100644
--- a/builtins/read.def
+++ b/builtins/read.def
@@ -326,6 +326,8 @@ read_builtin (list)
       code = setjmp (alrmbuf);
       if (code)
 	{
+	  interrupt_immediately--;
+	  terminate_immediately = 0;
 	  run_unwind_frame ("read_builtin");
 	  return (EXECUTION_FAILURE);
 	}
diff --git a/patchlevel.h b/patchlevel.h
index 37480dc0b2c6..f72e2d850936 100644
--- a/patchlevel.h
+++ b/patchlevel.h
@@ -25,6 +25,6 @@
    regexp `^#define[ 	]*PATCHLEVEL', since that's what support/mkversion.sh
    looks for to find the patch level (for the sccs version string). */
 
-#define PATCHLEVEL 50
+#define PATCHLEVEL 51
 
 #endif /* _PATCHLEVEL_H_ */
